tools/ports/__init__.py

```diff
@@ -57,13 +55,32 @@ def load_port(name):
   if not hasattr(port, 'variants'):
     # port variants (default: no variants)
     port.variants = {}
+  if not hasattr(port, 'show') and port.is_contrib:
```

Comment:
the auto generated `show` method relies on `port.project_license()` (or `port.license()`) which is required only for contrib ports. That being said if `port.license()` is not implemented then CI sanity checks will fail when it invokes `show_ports()` so it's safe to remove this check anyway because it would be caught